hubblestack_nova/modules/cve_scan_v2.py

```diff
@@ -0,0 +1,283 @@
+        affected_version = affected_version[:-4]
+
+    #Compare from higher order to lower order based on '-' split.
+    local_version_split = local_version.split('-')
+    affected_version_split = affected_version.split('-')
+
+    for order_index in range(len(local_version_split)):
+        
+        local_version_obj = LooseVersion(local_version_split[order_index])
+        affected_version_obj = LooseVersion(affected_version_split[order_index])
+        
+        #Check lower order bits if higher order are equal.
+        if local_version == affected_version:
+            continue
+
+        #Return when highest order version is not equal.
+        elif local_version_obj > affected_version_obj:
+            return False
+        elif local_version_obj < affected_version_obj:
+            return True
+        
+    else:
+        # The packages are equal if the code has gotten to here.
+        #     Now return based on the operator.
+        if operator == 'le':
+            return True
+        elif operator == 'lt':
+            return False        
+    
+
+
+
+def _get_cache(ttl, url):
+    '''
+    If url contains valid cache, returns it,
+        Else returns empty dictionary.
+    ###TODO### more effective caching method, return
+            historic data and only query for new vulnerabilites
+    '''
+
+    if url.startswith('salt://'):
+        path_to_json = url[len('salt://'):]
+        ########## TODO ##############
+    elif url.startswith('http://') or url.startswith('https://'):
+        # Check if we have a valid cached version.            
+        try:
+            cached_time = os.path.getmtime(cache_path)
+        except OSError:
+            return {}
+        if current_time() - cached_time < ttl:
+            try:
+                with open(cache_path) as json_file:
+                    loaded_json = json.load(json_file)
+                    return loaded_json
+            except IOError as e: 
+                return {}
+        else:
+            return {}
+
+def _build_json(master_json, current_page):
+    '''
+    Adds all the search elements from current page
+        to our master json file and returns
+    '''
+    current_page_search = current_page['data']['search']
+    master_json['data']['search'].extend(current_page_search)
+    return master_json
+
+
+class vulnerablePkg:
+    '''
+    Object representing a vulnverable pkg for the current operating system.
+    '''
+    def __init__(self, pkg, pkg_version, score, operator, reporter, href, cve_list):
+        self.pkg = pkg
+        self.pkg_version = pkg_version
+        self.pkg_version = pkg_version
+        self.score = score
+        if operator not in ['lt', 'le']:
+            log.error('pkg:%s contains an operator that\'s not supported and waschange to < ')
+            operator = 'lt'
+        self.operator = operator
+        self.href = href
+        self.cve_list = cve_list
+        self.reporter = reporter
+    def get_pkg(self):
```

Comment:
These "getter" functions are not necessary in python. You can reference internal variables externally just fine in python:

```
>>> class foo:
...   def __init__(self, bar):
...     self.bar = bar
...
>>> myfoo = foo('bar')
>>> myfoo.bar
'bar'
```